sycamore/docset.py

```diff
@@ -377,18 +377,97 @@ def extract_entity(self, entity_extractor: EntityExtractor, **kwargs) -> "DocSet
         return DocSet(self.context, entities)
 
     def extract_schema(self, schema_extractor: SchemaExtractor, **kwargs) -> "DocSet":
+        """
+        Extracts a JSON schema of extractable properties from each document in this DocSet.
+
+        Each schemais a mapping of names to types that corresponds to fields that are present in the document.
+        For example, calling this method on a financial document containing information about companies
+        might yield a schema like
+
+        .. code-block:: python
+
+            {
+              "company_name": "string",
+              "revenue": "number",
+              "CEO": "string"
+            }
+
+        This method will extract a unique schema for each document in the DocSet independently.
+        If the documents in the DocSet represent instances with a common schema, consider
+        `ExtractBatchSchema` which will extract a common schema for all documents.
+
+        The dataset is returned with an additional `_schema` property that contains JSON-encoded schema, if any
+        is detected.
+
+        Args:
+            schema_extractor: A `SchemaExtractor` instance to extract the schema for each document.
+
+        Example:
+            .. code-block:: python
+
+                openai_llm = OpenAI(OpenAIModels.GPT_3_5_TURBO.value)
+                schema_extractor=OpenAISchemaExtractor("Corporation", llm=openai, num_of_elements=35)
+
+                context = sycamore.init()
+                pdf_docset = context.read.binary(paths, binary_format="pdf")
+                    .partition(partitioner=UnstructuredPdfPartitioner())
+                    .extract_schema(schema_extractor=schema_extractor)
+        """
+
         from sycamore.transforms import ExtractSchema
 
         schema = ExtractSchema(self.plan, schema_extractor=schema_extractor)
         return DocSet(self.context, schema)
 
     def extract_batch_schema(self, schema_extractor: SchemaExtractor, **kwargs) -> "DocSet":
+        """
+        Extracts a common schema from the documents in this DocSet.
+
+        This transform is similar to extract_schema, except that it will add the same schema
+        to each document in the DocSet rather than infering a separate schema per Document.
+        This is most suitable for document collections that share a common format. If you have
+        a heterogeneous document collection, consider using extract_schema instead.
```

Comment:
Should we make this suggestion? It might be hard to use this metadata if it's different for each doc. Could write "If you have a heterogeneous document collection and want a different schema for each type..."

Happy leaving it in though, if you think otherwise.